etcd-manager/pkg/volumes/scaleway/volumes.go

```diff
@@ -0,0 +1,226 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scaleway
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/etcdadm/etcd-manager/pkg/volumes"
+)
+
+const (
+	localDevicePrefix = "/dev/disk/by-id/scsi-0SCW_b_ssd_volume-"
+)
+
+// ScwVolumes defines the Scaleway Cloud volume implementation.
+type ScwVolumes struct {
+	clusterName string
+	matchTags   []string
+	nameTag     string
+
+	scwClient   *scw.Client
+	server      *instance.Server
+	zone        scw.Zone
+	instanceAPI *instance.API
+}
+
+var _ volumes.Volumes = &ScwVolumes{}
+
+// NewScwVolumes returns a new Scaleway Cloud volume provider.
+func NewScwVolumes(clusterName string, volumeTags []string, nameTag string) (*ScwVolumes, error) {
+	scwClient, err := scw.NewClient(
+		scw.WithEnv(),
+	)
+	if err != nil {
+		return nil, err
```

Comment:
Nit: I generally suggest wrapping errors when they come from code outside of "our" projects (of course you might consider the scaleway SDK one of "our" projects, and if the errors are self-descriptive, then it's fine _not_ to wrap).  The AWS SDK errors were/are pretty terse, so without wrapping they can be hard to decipher.